addon/active-model-serializer.js

```diff
@@ -300,6 +300,24 @@ var ActiveModelSerializer = RESTSerializer.extend({
       }, this);
     }
   },
+
+  extractRelationships: function(modelClass, resourceHash) {
+    modelClass.eachRelationship(function (key, relationshipMeta) {
+      var relationshipKey = this.keyForRelationship(key, relationshipMeta.kind, "deserialize");
+      if (resourceHash.hasOwnProperty(relationshipKey)) {
+        var polymorphicTypeKey = this.keyForRelationship(key) + '_type';
```

Comment:
`keyForRelationship` basically decamelize if you pass only a key, that's exactly what I wanted here. I choose this in order to add the minimum code as possible.

You're right about the additionnal check, will add a failing test first :)